numba/tests/test_numba_array.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import print_function, division, absolute_import
+
+import numpy as np
+
+from numba import unittest_support as unittest
+from numba import jit, types, errors, typeof, numpy_support, cgutils
+from numba.compiler import compile_isolated
+from .support import TestCase, captured_stdout
+from numba.array import Array
+
+def len_usecase(x):
+    return len(x)
+
+def print_usecase(x):
+    print(x)
+
+def getitem_usecase(x, key):
+    return x[key]
+
+def shape_usecase(x):
+    return x.shape
+
+def npyufunc_usecase(x):
+    return np.cos(np.sin(x))
+
+
+def identity(x): return x
+
+class TestArray(TestCase):
```

Comment:
Perhaps split the test for JIT usage and Python-facing usage into two separate classes?